sklearn/metrics/ranking.py

```diff
@@ -889,3 +889,178 @@ def label_ranking_loss(y_true, y_score, sample_weight=None):
     loss[np.logical_or(n_positives == 0, n_positives == n_labels)] = 0.
 
     return np.average(loss, weights=sample_weight)
+
+
+def gini_coefficient_score(y_true, y_score, sample_weight=None):
+    """Compute the Gini coefficient from prediction scores.
+
+    Note: Should not be confused with Gini Impurity
+
+    Parameters
+    ----------
+    y_true : array, shape = [n_samples]
+        True responses.
+
+    y_score : array, shape = [n_samples]
+        Target (ranking) scores.
+
+    sample_weight : array-like of shape = [n_samples], optional
+        Sample weights.
+
+    Returns
+    -------
+    gini_coefficient : float
+        Gini coefficient. Corresponds to the area between the the diagonal and
+        the Lorenz curve
+
+    References
+    ----------
+    .. [1] `Wikipedia entry for the Gini Coefficient
+            <https://en.wikipedia.org/wiki/Gini_coefficient>`_
+
+    See also
+    --------
+
+    lorenz_curve : Compute Lorenz curve
+
+    normalized_gini_coefficient_score: Normalized Gini coefficient
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.metrics import gini_coefficient_score
+    >>> y_true = np.array([0.0, 2.0, 2.5, 5.4])
+    >>> y_scores = np.array([0.1, 0.4, 0.35, 0.8])
+    >>> gini_coefficient_score(y_true, y_scores)
+    0.19823232323232332
+
+    """
+    cumulative_observations, cumulative_responses = \
+        lorenz_curve(y_true, y_score, sample_weight=sample_weight,
+                      normalized=True)
+    gini_coefficient = auc(cumulative_observations, cumulative_responses) - 0.5
```

Comment:
Isn't it supposed to be multiplied by 2 to span the [0-1] range on average?

```suggestion
    gini_coefficient = 2 * auc(cumulative_observations, cumulative_responses) - 1
```